src/impl/epoll/external_commit_helper.hpp

```diff
@@ -53,22 +57,38 @@ class ExternalCommitHelper {
         FdHolder(FdHolder const&) = delete;
     };
 
-    void listen();
+    class DaemonThread {
```

Comment:
Something like https://gist.github.com/bdash/9fab3dbe65449b324ae773adb6fc74c5 is what I had in mind.